cluster.go

```diff
@@ -27,192 +26,152 @@ import (
 )
 
 const (
-	marathonNodeUp   = 0
-	marathonNodeDown = 1
+	memberStatusUp   = 0
+	memberStatusDown = 1
 )
 
-// Cluster is the interface for the marathon cluster impl
-type Cluster interface {
-	URL() string
-	// retrieve a member from the cluster
-	GetMember() (string, error)
-	// make the last member as down
-	MarkDown()
-	// the size of the cluster
-	Size() int
-	// the members which are available
-	Active() []string
-	// the members which are NOT available
-	NonActive() []string
-}
+// the status of a member node
+type memberStatus int
 
-type marathonCluster struct {
+// cluster is a collection of marathon nodes
+type cluster struct {
 	sync.RWMutex
-	// the cluster url
-	url string
-	// a link list of members
-	members *marathonNode
-	//  the number of members
-	size int
-	// the protocol
-	protocol string
-	// the current host
-	active *marathonNode
+	// a collection of nodes
+	members []*member
 	// the http client
 	client *http.Client
 }
 
-// String returns a string representation of the cluster
-func (r *marathonCluster) String() string {
-	return fmt.Sprintf("url: %s|%s, members: %s, size: %d, active: %s",
-		r.protocol, r.url, r.members, r.size, r.active)
-}
-
-type marathonNode struct {
+// member represents a individual endpoint
+type member struct {
 	// the name / ip address of the host
-	hostname string
+	endpoint string
 	// the status of the host
-	status int
-	// the next member in the list
-	next *marathonNode
+	status memberStatus
 }
 
-func (member marathonNode) String() string {
-	status := "UP"
-	if member.status == marathonNodeDown {
-		status = "DOWN"
-	}
+// newCluster returns a new marathon cluster
+func newCluster(client *http.Client, marathonURL string) (*cluster, error) {
+	// step: extract and basic validate the endpoints
+	var members []*member
+	var defaultProto string
 
-	return fmt.Sprintf("member: %s:%s", member.hostname, status)
-}
+	for _, endpoint := range strings.Split(marathonURL, ",") {
+		// step: check for nothing
+		if endpoint == "" {
+			return nil, ErrInvalidEndpoint
```

Comment:
We could improve the user experience here by turning `ErrInvalidEndpoint` into a struct that takes an error description. That way, users would know what kind of error they produced.

Could also happen in a separate PR. Leave that up to you.